shared_code/base_scraper.py

```diff
@@ -212,6 +292,9 @@ def main_scraper(self, client: MongoClient) -> None:
             logger.info(f'final df shape: {df.shape}')
             self.add_required_fields(df)
             if len(df) > 0:
+                # Only add the data if the dataframe contains each of the following fields: name, address1, city, state, zip
+                if not self.validate_data(df):
```

Comment:
Raise an Exception whenever you want to exit the program instead of silently returning and let downstream applications decide what do with that exception. I am raising exception here because then azure function code in `init.py` can catch it and mark it as a failure to correctly update the telemetry for that azure function.